src/Framework/BuildEngineInterface.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+
+namespace Microsoft.Build.Framework
+{
+    /// <summary>
+    /// Exposes build engine functionality that was made available in newer versions of MSBuild.
+    /// </summary>
+    /// <remarks>
+    /// Make all members virtual but not abstract, ensuring that implementations can override them and external implementations
+    /// won't break when the class is extended with new members. This base implementation should be throwing <see cref="NotImplementedException"/>.
+    /// </remarks>
+    [Serializable]
+    public abstract class BuildEngineInterface
+    {
+        /// <summary>
+        /// Initial version with LogsMessagesOfImportance() as the only exposed member.
+        /// </summary>
+        public const int Version1 = 1;
```

Comment:
I'm not sure I understand the need for this member, can you elaborate?

Reply:
The idea is that the versioning history will be captured in these constants. I know it's lame but having the `Version` prop return 5 is not super readable without describing how it is different from 4. And while digging this information from git history would work for us internally, we should also have something in the external doc as this is public API.